src/zerocoin.cpp

```diff
@@ -528,17 +528,33 @@ bool CheckZerocoinTransaction(const CTransaction &tx,
                               bool fStatefulZerocoinCheck,
                               CZerocoinTxInfo *zerocoinTxInfo)
 {
+    // nHeight have special mode which value is INT_MAX so we need this.
+    int realHeight;
+
+    {
+        LOCK(cs_main);
+        realHeight = chainActive.Height();
+    }
+
     // Check Mint Zerocoin Transaction
-    BOOST_FOREACH(const CTxOut &txout, tx.vout) {
+    for (const CTxOut &txout : tx.vout) {
         if (!txout.scriptPubKey.empty() && txout.scriptPubKey.IsZerocoinMint()) {
+            if (realHeight > params.nSigmaStartBlock + params.nZerocoinV2MintGracefulPeriod) {
```

Comment:
I intentionally did not put this check here. The problem is this function is called both from ConnectBlock and AcceptToMemoryPoolWorker. And there are already checks for this. We needed 2 different checks, one for mempool addition, another for blocks.